### DISCUSSION TITLE
Build fails because of the valarray include

### DISCUSSION THREAD

**Discussion by ThiloGHL** [nlohmann/json#3470] [Q&A] [answered]
Hi,

when including the json.hpp file in the header file, the build fails with following errors:
![errors](https://user-images.githubusercontent.com/62743504/166891333-c50ddb96-a1db-41a1-a956-293c35c343e2.png)

It's because of the valarray includes. If I remove the valarray code, it works.
When including the file in the cpp file, it works too. But I need it in the header file for the method prototypes. 

How can I solve this problem?

Thank you,

Thilo

**Comment by nlohmann** [comment, marked as answer]
Which compiler are you using?

**Reply by ThiloGHL** [reply]
Visual Studio 2019
Microsoft (R) C/C++-Optimierungscompiler Version 19.29.30143 für x64
C++ Version is v14

**Reply by nlohmann** [reply]
That is very strange as we test the library successfully with multiple MSVC versions (using C++11, C++14, and C++17):

<img width="841" alt="image" src="https://user-images.githubusercontent.com/159488/166894982-caa52c9b-797a-4a38-97e6-14e4dd7a5321.png">

Can you compile and run the unit tests?

**Reply by ThiloGHL** [reply]
Sorry, I have never run a unit test before and don't know how I have to proceed.

**Reply by nlohmann** [reply]
See https://github.com/nlohmann/json#execute-unit-tests

**Reply by ThiloGHL** [reply]
Thank you, but where do I have to execute this? Do I need to install CMake?

**Reply by nlohmann** [reply]
Yes, you need CMake.

How did you build your example?

**Reply by nlohmann** [reply]
Do you need further assistance with this issue?

**Reply by ThiloGHL** [reply]
Sorry, I didn't get to it last week and forgot to reply. I'll try to get to it again this week.
I built the example using Visual Studion 2019, but the VS 2017 (v141) platform toolset.

**Reply by nlohmann** [reply]
@ThiloGHL Do you need further assistance with this issue?

**Reply by ThiloGHL** [reply]
Sorry, I was on vacation and completely forgot about this. Unfortunately I won't get to run a unit test anytime soon, but currently we can use the library with modifications.
So this discussion can be closed. Thanks.
